src/operator/nn/dnnl/dnnl_eltwise-inl.h

```diff
@@ -50,17 +52,18 @@ class DNNLTanhFwd {
   std::shared_ptr<eltwise_fwd_pd_t> fwd_pd;
 };
 
-inline void DNNLTanhForward(const nnvm::NodeAttrs& attrs,
-                            const OpContext& ctx,
-                            const NDArray& input,
-                            const OpReqType& req,
-                            const NDArray& output) {
-  DNNLTanhFwd& fwd = DNNLTanhFwd::GetTanhForward(input, output);
+template <dnnl::algorithm algorithm>
+inline void DNNLEltwiseForward(const nnvm::NodeAttrs& attrs,
```

Comment:
Any reason why this is a template?

Reply:
DNNLRun function takes only 5 parameters and using template allows for sixth.